src/components/Voter.jsx

```diff
@@ -1,7 +1,31 @@
-export default function Voter({ votes, }) {
+import { useState, } from "react";
+import api from "../api";
+
+
+export default function Voter({ requestConf, initialVotes }) {
+  requestConf = structuredClone(requestConf);
+  requestConf.method = "PATCH";
+  const[votes, setVotes] = useState(initialVotes);
   function handleVote(event) {
-    alert(`This button doesn't do anything yet!
-      Button: ${event.target.name}`);
+    if(event.target.name === "down") {
+      requestConf.data = {
+        inc_votes: -1,
+      };
```

Comment:
This approach of building up the requestConfig does work - but would it be possible to extract some of this over into your api.js to reduce the clutter and simplify the component file a bit?

You could write a function designed to make that request, and invoke it with an inc_votes of either -1 or 1 to decide between an upvote and a downvote.